### PR TITLE
[provisioning] Integrate local provisioning orchestrator with remote database

### PR DESCRIPTION
**Pull request by nmoroze**
This PR adds functionality to integrate the local provisioning orchestrator script with a Google Cloud Firestore database. It adds a new script called `update_remote_db.py`, updates the orchestrator itself, and adds documentation in the form of a README and `requirements.txt` file. 

The new orchestrator workflow is as follows:
- After running the provisioning flow, the orchestrator still writes the device record to a local SQLite database 
- Afterwards, the orchestrator will attempt to upload *all* records in the local SQLite database, deleting each one from the local DB on success
- If records remain in the local DB after provisioning, a message is printed indicating to the user that they need to run `update_remote_db.py` to upload the additional records

`update_remote_db.py` takes in a path to the local database and will attempt to upload all records to the remote DB, removing each from the local DB on success. 

In both cases, if a duplicate entry is detected in the remote database, the user will be prompted with the ability to skip the entry, overwrite the remote entry, or remove the entry from the local DB. 

The Firestore database structure is fairly simple. It contains a single "collection" called `devices`, which contains "documents" keyed by device ID. Each document is a flat structure with keys and values that map directly to the SQL schema used for the local database.

### CODE REVIEW

File: sw/host/provisioning/orchestrator/update_remote_db.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    parser = argparse.ArgumentParser(
+        prog="Remote Database Update",
+        description="This tool updates the ES provisioning remote database with "
+                    "locally cached device registrations.",
+    )
+    parser.add_argument("db_path",
```
Comment (milesdai) [inline sw/host/provisioning/orchestrator/update_remote_db.py:112]: Is `db_path` also a required argument?

**Review by milesdai — APPROVED** [review]
LGTM! Thanks!

**Review by timothytrippel — APPROVED** [review]

File: sw/host/provisioning/orchestrator/update_remote_db.py (hunk 1 above)
Comment (nmoroze) [reply sw/host/provisioning/orchestrator/update_remote_db.py:112]: Positional arguments are required by default (and specifying `required` results in an error).

File: sw/host/provisioning/orchestrator/update_remote_db.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    local_db = sqlite3.connect(args.db_path)
+
+    try:
+        update_remote_db(args.project, local_db)
+    except DefaultCredentialsError:
+        print(
+            "Not authenticated to gcloud. Please run "
+            "`gcloud auth application-default login` to generate credentials"
+        )
```
[the remark this answers is not in the document]
Comment (nmoroze) [reply sw/host/provisioning/orchestrator/update_remote_db.py:127]: Yeah, nice catch - updated!

**Comment by timothytrippel** [comment]
We can ignore the kokoro error here just FYI.

**Pull request merged**